lib/rex/socket.rb

```diff
@@ -939,26 +940,42 @@ def self.rex_gethostbyname(name, resolver: @@resolver)
   # @param resolver [Rex::Proto::DNS::CachedResolver] Resolver to query for the name
   # @return [Array] Array mimicking the native getaddrinfo return type
   def self.rex_getaddrinfo(name, resolver: @@resolver)
-    v4, v6 = self.rex_resolve_hostname(name, resolver: resolver)
-    # Build response array
     getaddrinfo = []
-    v4.each do |a4|
+
+    if name =~ /^\d+$/ && name.to_i.between?(0, 0xffffffff)
```

Comment:
Not a blocker; `\A` and `\z` are usually used for anchors in Ruby https://stackoverflow.com/questions/3632024/why-do-rubys-regular-expressions-use-a-and-z-instead-of-and